Tasks/Common/webdeployment-common/fileTransformationsUtility.ts

```diff
@@ -7,9 +7,18 @@ var jsonSubstitutionUtility = require('webdeployment-common/jsonvariablesubstitu
 var xmlSubstitutionUtility = require('webdeployment-common/xmlvariablesubstitutionutility.js');
 var xdtTransformationUtility = require('webdeployment-common/xdttransformationutility.js');
 
-export function fileTransformations(isFolderBasedDeployment: boolean, JSONFiles: any, xmlTransformation: boolean, xmlVariableSubstitution: boolean, folderPath: string) {
+export function fileTransformations(isFolderBasedDeployment: boolean, JSONFiles: any, xmlTransformation: boolean, xmlVariableSubstitution: boolean, folderPath: string, webDeployPkg: string) {
 
     if(xmlTransformation) {
+        if(!isFolderBasedDeployment && utility.isMSDeployPackage(webDeployPkg)) {
+            var debugMode = tl.getVariable('system.debug');
+            if(debugMode && debugMode.toLowerCase() == 'true') {
+                tl.warning(tl.loc('AutoParameterizationMessage'));
```

Comment:
Message is available in [AzureRmWebAppDeployment/task.json](https://github.com/Microsoft/vsts-tasks/blob/73df31463d98bdcde73c5441edeed76f83371f8a/Tasks/AzureRmWebAppDeployment/task.json#L481)